process/notifier/sovereignNotifier.go

```diff
@@ -126,77 +135,50 @@ func checkNilOutportBlockFields(outportBlock *outport.OutportBlock) error {
 	return nil
 }
 
-func (notifier *sovereignNotifier) createAllIncomingMbs(txPool *outport.TransactionPool) ([]*block.MiniBlock, error) {
-	mbs := make([]*block.MiniBlock, 0)
+func (notifier *sovereignNotifier) createIncomingEvents(logsData []*outport.LogData) []*transaction.Event {
+	incomingEvents := make([]*transaction.Event, 0)
 
-	txsMb, err := notifier.createIncomingMbsFromTxs(txPool.Transactions)
-	if err != nil {
-		return nil, err
+	for _, logData := range logsData {
+		events := logData.GetLog().Events
+		eventsFromLog := notifier.getIncomingEvents(events, logData.TxHash)
+
+		incomingEvents = append(incomingEvents, eventsFromLog...)
 	}
 
-	// TODO: when specs are defined, we should also handle scrs mbs
-	mbs = append(mbs, txsMb...)
-	return mbs, nil
+	return incomingEvents
 }
 
-func (notifier *sovereignNotifier) createIncomingMbsFromTxs(txs map[string]*outport.TxInfo) ([]*block.MiniBlock, error) {
-	execOrderTxHashMap := make(map[string]uint32)
-	shardIDTxHashMap := make(map[uint32][][]byte)
+func (notifier *sovereignNotifier) getIncomingEvents(events []*transaction.Event, txHash string) []*transaction.Event {
+	incomingEvents := make([]*transaction.Event, 0)
 
-	for txHash, tx := range txs {
-		receiver := tx.GetTransaction().GetRcvAddr()
-		_, found := notifier.subscribedAddresses[string(receiver)]
-		if !found {
+	for _, event := range events {
+		if !notifier.isSubscribed(event, txHash) {
 			continue
 		}
 
-		hashBytes, err := hex.DecodeString(txHash)
-		if err != nil {
-			return nil, fmt.Errorf("%w, hash: %s", err, txHash)
-		}
-
-		log.Info("found incoming tx", "tx hash", txHash, "receiver", hex.EncodeToString(receiver))
-
-		execOrderTxHashMap[string(hashBytes)] = tx.GetExecutionOrder()
-
-		sender := tx.GetTransaction().GetSndAddr()
-		senderShardID := notifier.shardCoordinator.ComputeId(sender)
-		shardIDTxHashMap[senderShardID] = append(shardIDTxHashMap[senderShardID], hashBytes)
+		incomingEvents = append(incomingEvents, event)
 	}
 
-	return createSortedMbs(execOrderTxHashMap, shardIDTxHashMap, block.TxBlock), nil
+	return incomingEvents
 }
 
-func createSortedMbs(
-	execOrderTxHashMap map[string]uint32,
-	shardIDTxHashMap map[uint32][][]byte,
-	mbType block.Type,
-) []*block.MiniBlock {
-	mbs := make([]*block.MiniBlock, 0, len(shardIDTxHashMap))
+func (notifier *sovereignNotifier) isSubscribed(event *transaction.Event, txHash string) bool {
+	for _, subEvent := range notifier.subscribedEvents {
```

Comment:
could this be optimized with check in a map? 
`if eventsMap.contains(event.GetIdentifier()) `

Reply:
We will most likely have max 2-3 subscribed events.
This could be changed to use an events map, but would introduce overhead in case of the following scenario:

- Suppose we introduce another field: `Topics` in `SubscribedEvent` struct and have the following config:

```
subscribed_events = [
    { identifier = "deposit", topic = "topic1", addresses = ["erd1a", "erd1b"] },
    { identifier = "deposit", topic = "topic2", addresses = ["erd1c", "erd1d"] }
]
```

If we use a map[identifier] then those two keys would collide.
I would leave it as it is

Reply:
ok! you could have maps all the way down but if there will be only 2-3 events, it should work also like this.